From 834112fa6357d5c603239f5e41af2f63560d667b Mon Sep 17 00:00:00 2001
From: Gwenael Treuveur <gwenael.treuveur@foss.st.com>
Date: Tue, 29 Oct 2024 10:30:54 +0100
Subject: [PATCH 1380/1695] dts: stm32: add OEM rproc encryption key for
 stm32mp25

Add new OTP entry for remoteproc encryption key.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 069792babb071f984e3947d6d250ad667d81b2a8 dts: stm32: add OEM rproc encryption key for stm32mp25]
Change-Id: I11268ecb339eed5adc9aba6f84483d03abda776d
Signed-off-by: Gwenael Treuveur <gwenael.treuveur@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/415026
Reviewed-by: Gwenael TREUVEUR <gwenael.treuveur@st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Tested-by: Gwenael TREUVEUR <gwenael.treuveur@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 core/arch/arm/dts/stm32mp251.dtsi | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp251.dtsi b/core/arch/arm/dts/stm32mp251.dtsi
index a2600ac33..eea0d23b6 100644
--- a/core/arch/arm/dts/stm32mp251.dtsi
+++ b/core/arch/arm/dts/stm32mp251.dtsi
@@ -2923,6 +2923,12 @@
 				st,non-secure-otp-provisioning;
 			};
 
+			oem_rproc_enc_key: oem_rproc_enc_key@540 {
+				/* 0x540 -> otp336 */
+				reg = <0x540 0x20>;
+				st,non-secure-otp-provisioning;
+			};
+
 			oem_enc_key_2: otp360@5a0 {
 				reg = <0x5a0 0x10>;
 				st,non-secure-otp-provisioning;
-- 
2.47.2

